libcaliptra/examples/generic/test.c

```diff
@@ -668,6 +668,96 @@ int rom_test_devid_csr(const test_info* info)
     return failure;
 }
 
+// Issue FW load commands repeatedly
+// Coverage for piecewise FW load and runtime FW updates
+int upload_fw_piecewise(const test_info* info)
+{
+    int failure = 0;
+    int status = boot_to_ready_for_fw(info, false);
+
+    if (status){
+        printf("Failed to boot to ready for FW: 0x%x\n", status);
+        failure = 1;
+    }
+
+    // Some "random" size to split up the FW load into chunks
+    // These represent the first two chunks, the third chunk is the remainder of the image
+    // Sizes of 0 are ignored (meaning one fewer chunk is sent)
+    uint32_t chunk_sizes[][2] =    {
```

Comment:
Ideally the chunk should be randomly generated, and make sure they don`t add up for more space than the total image size...but I understand this is non production code...so it is not a sticky point

Reply:
I think the methodology I have seen in Caliptra is for deterministic testing over randomization which can cause flakeyness. I do agree that would be better coverage overall tough. (So long as the corner cases are for the code being tested and not corner cases in the test itself.)